src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,139 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { Fragment, ReactElement, ReactNode, useState } from 'react';
+import { useDrag } from 'react-dnd';
+
+import { DropZone } from '@components/DropZone';
+import { TreeItemProps } from '@components/Tree/types';
+import { useTreeContext } from '@components/Tree/TreeContext';
+import { DraggableItem, DropZonePosition } from '@utilities/dnd';
+import { merge } from '@utilities/merge';
+
+export const TreeItem = ({ id, sort, onNodeSelect, onDrop, component, children }: TreeItemProps) => {
```

Comment:
`onNodeSelect` rename to `onSelect`. And then do `onSelect: OnItemSelect` to avoid name collision.